types/context.go

```diff
@@ -275,6 +275,21 @@ func (c Context) CacheContext() (cc Context, writeCache func()) {
 	return cc, cms.Write
 }
 
+// CacheContextEmitEvents behaves the same as CacheContext except that it
```

Comment:
Why not include in cache context?

Reply:
WDYM?

Reply:
I think the question is why the choice to be additive instead of modifying the existing `CacheContext`?  I'm also wondering.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/13063#issuecomment-1231155644